operator/controllers/agent/configmaps.go

```diff
@@ -38,7 +40,48 @@ import (
 )
 
 // configMapForAgentConfig prepares the ConfigMap object for the Agent.
-func configMapForAgentConfig(instance *agentv1alpha1.Agent, scheme *runtime.Scheme) (*corev1.ConfigMap, error) {
+func configMapForAgentConfig(
+	ctx context.Context,
+	client client.Client,
+	instance *agentv1alpha1.Agent,
+	scheme *runtime.Scheme,
+) (*corev1.ConfigMap, error) {
+	// HACK: auto-inject controller's cert to agent if we detect it connects to local controller.
+	// FIXME: Mounting optional configmap with controller CA would be a better
+	// solution, but right now we don't have a configmap with CA, just a secret
+	// that contains also the key.
+	// Also, we should figure out the way so that both CA file from config and
+	// controller's CA can be used simultaneously.
+	// FIXME: Even better solution would be for local controller to be signed
+	// using https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/
+	var localControllerCert []byte
+	for _, endpoint := range instance.Spec.ConfigSpec.AgentFunctions.Endpoints {
+		if !isLocalControllerEndpoint(endpoint) {
+			continue
+		}
+
+		caFile := &instance.Spec.ConfigSpec.AgentFunctions.ClientConfig.GRPCClient.ClientTLSConfig.CAFile
+		if *caFile != "" {
+			continue
+		}
+
+		if client == nil {
+			continue
+		}
+
+		var secret corev1.Secret
+		_ = client.Get(
+			ctx,
+			types.NamespacedName{Namespace: "aperture-controller", Name: "controller-controller-cert"},
```

Comment:
The namespace and the name could be different based on the installation done by user so we shouldn't use it this way.